### PR TITLE
Fix NPE for taskGroupId when rolling update

### PR DESCRIPTION
**Pull request by jihoonson**
Fixes https://github.com/apache/incubator-druid/issues/6163.

### CODE REVIEW

File: indexing-service/src/main/java/io/druid/indexing/overlord/supervisor/SupervisorManager.java [hunk 1]
```diff
@@ -165,7 +165,8 @@ public boolean resetSupervisor(String id, @Nullable DataSourceMetadata dataSourc
 
   public boolean checkPointDataSourceMetadata(
       String supervisorId,
-      int taskGroupId,
+      @Nullable String taskId,
+      Integer taskGroupId,
```
Comment (nishantmonu51) [inline indexing-service/src/main/java/io/druid/indexing/overlord/supervisor/SupervisorManager.java]: should taskGroupId be nullable instead of taskId ?

Comment (jihoonson) [reply indexing-service/src/main/java/io/druid/indexing/overlord/supervisor/SupervisorManager.java]: Thanks. Fixed.

**Comment by pdeva** [comment]
note that this issue can occur well after the rolling update. we are seeing this on our fully updated cluster

**Comment by jihoonson** [comment]
@pdeva this error should never happen once the cluster is fully updated. Do you have the overlord and task logs? If you have, please add them to #6163.

**Comment by pdeva** [comment]
replied on drudi-dev thread. they have recovered for now but will indeed post logs if it occurs again.

**Comment by jihoonson** [comment]
@pdeva thanks.

BTW, this patch is incomplete. I think we need a way to find `taskGroup` using `baseSequenceName` to fully support rolling update. I'll update this PR soon.

**Comment by fjy** [comment]
👍

**Pull request merged**